### DISCUSSION TITLE
Headless GCS

### DISCUSSION THREAD

**Discussion by fponticelli** [richardwilkes/gcs#680] [Ideas]
How hard would it be to make a version of GCS that is headless? By that I mean, that the business logic is fully separated from the UI. If that is not too difficult, GCS Headless could be compiled to a WASM binary, allowing to be used directly in a web page without using a web server.
I guess another consideration would be around abstracting asset loading.

**Comment by fponticelli** [comment]
Potentially even the UI could be ported by drawing directly to a Canvas element potentially even using WebGL.

**Comment by richardwilkes** [comment]
The model is already largely separate from the UI in the current code base.

Compiling to WASM is fairly straight-forward with Go... however, you'd still need data access (currently the local file system) as well as a *lot* of UI code to do anything useful with it.

**Comment by fponticelli** [comment]
Is the data access abstracted already? That would have to pass through JS. Just being able to embed GCS Model in VTT and interact directly with the existing Character Sheet would be killer. @crnormand FYI

**Comment by richardwilkes** [comment]
No, it isn't
